### ISSUE TITLE
Cannot declare class with forward references superclass

### ISSUE THREAD

**Issue by drmeister** [drmeister/clasp#130]
`(defclass foo (undefined-class) ())` fails.

It has something to do with:  standard.lisp ... the (finalize-inheritance class) ???

```
(defmethod allocate-instance ((class class) &rest initargs)
  (declare (ignore initargs))
  ;; FIXME! Inefficient! We should keep a list of dependent classes.
  (unless (class-finalized-p class)
    (finalize-inheritance class))
  #+clasp(unless *the-class-class*
       (setq *the-class-class* (find-class 'class)))
  (let ((x #-clasp(si::allocate-raw-instance nil class (class-size class))
       #+clasp(if (core:subclassp class *the-class-class*)
              (ALLOCATE-RAW-CLASS nil class (class-size class))
              (si::allocate-raw-instance nil class (class-size class)))
       ))
    (si::instance-sig-set x)
    x))
```

**Comment by Shinmera** [comment]
Confirmed on b05884f
```
> (defclass foo (bar) ())
;;; Warning: Class BAR has been forward referenced.

Condition of type: SIMPLE-TYPE-ERROR
In lisp_errorUnexpectedType src/core/foundation.cc line 161
COMMON-LISP:TYPE-ERROR :initializers (:DATUM #S(<ADD SUPPORT FOR INSTANCE _CLASS=COMMON-LISP:NIL > #slots[0]
)
 :EXPECTED-TYPE COMMON-LISP:CLASS  )
```

**Comment by Shinmera** [comment]
Almost fixed on b8c6690535bdc61bfdaeb50dbe23cd89633200a5
```
> (defclass foo (bar) ())
;;; Warning: Class BAR has been forward referenced.

#<STANDARD-CLASS FOO>
> (defclass bar () ())

Condition of type: TYPE-ERROR
#<FORWARD-REFERENCED-CLASS BAR> is not of type INSTANCE.
```

**Comment by Bike** [comment]
Can't reproduce on 35f15d4
